syntax = "proto3";

package com.chettoy.apexsky.apexlegends;

import "com/chettoy/apexsky/apexlegends/player.proto";

message AimKeyState {
  int32 aimbot_hotkey_1 = 1;
  int32 aimbot_hotkey_2 = 2;
  int32 attack_button = 3;
  int32 zoom_button = 4;
  int32 triggerbot_hotkey = 5;
  int32 attack_state = 6;
  int32 quick_looting_hotkey = 7;
}

message AimTargetInfo {
  float fov = 1;
  float distance = 2;
  bool is_visible = 3;
  bool is_knocked = 4;
  int32 health_points = 5;
  LoveStatusCode love_status = 6;
  bool is_kill_leader = 7;
  uint64 entity_ptr = 8;
  bool is_npc = 9;
  bool is_loot = 10;
}

message AimResultData {
  bool hitscan = 1;
  float view_pitch = 2;
  float view_yaw = 3;
  float delta_pitch = 4;
  float delta_yaw = 5;
  float delta_pitch_min = 6;
  float delta_pitch_max = 7;
  float delta_yaw_min = 8;
  float delta_yaw_max = 9;
  float distance = 10;
  Vec3 hitscan_nearest_pos = 11;
}
